File: backend/app/models/tenant.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base


class Tenant(Base):
    __tablename__ = "tenants"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    phone = Column(String(20), nullable=False, unique=True)
    id_card = Column(String(50))
    emergency_contact = Column(String(100))
    emergency_phone = Column(String(20))
    address = Column(String(500))
    note = Column(Text, default="")
    
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    contracts = relationship("Contract", back_populates="tenant")

File: frontend/src/lib/api/tenant.ts
import { api } from './client';

export interface Tenant {
  id: number;
  name: string;
  phone: string;
  id_card?: string;
  emergency_contact?: string;
  emergency_phone?: string;
  address?: string;
  note?: string;
  created_at: string;
  updated_at: string;
}

export interface TenantCreate {
  name: string;
  phone: string;
  id_card?: string;
  emergency_contact?: string;
  emergency_phone?: string;
  address?: string;
  note?: string;
}

export const tenantApi = {
  list: () => api.get<Tenant[]>('/tenants'),
  get: (id: number) => api.get<Tenant>(`/tenants/${id}`),
  create: (data: TenantCreate) => api.post<Tenant>('/tenants', data),
  update: (id: number, data: Partial<TenantCreate>) =>
    api.put<Tenant>(`/tenants/${id}`, data),
  remove: (id: number) => api.del(`/tenants/${id}`),
};
